### PR TITLE
sof-hda-dsp: add initial kcontrol values

### PR DESCRIPTION
**Pull request by libinyang**
This patch adds the initial values of sof-hda-dsp card.

Signed-off-by: Libin Yang <libin.yang@intel.com>

### CODE REVIEW

**Comment by plbossart** [comment]
@libinyang can you look at the errors and add the configurations to alsa-tests so that the ucm-validator works?

**Comment by libinyang** [comment]
> @libinyang can you look at the errors and add the configurations to alsa-tests so that the ucm-validator works?

Yes. I will do it. Just a silly question: How can I get the information in json file of the card, such as:
```
{
    "sof-hda-dsp-1": {
        "comment":	"Dummy test configuration for sof-hda-dsp",
        "id":		"sof-hda-dsp",
        "driver":	"sof-hda-dsp",
        "name":		"sof-hda-dsp",
        "longname":	"sof-hda-dsp",
        "components":   "cfg-dmics:2",
        "control": [
            { "id": "iface=CARD,name='Headphone Mic Jack'" },
            { "id": "name='Input Source'" }
        ]
    }
}
```

**Comment by libinyang** [comment]
And it seems the platform I use is not released. I will submit the patch to alsa-tests after the platform is released

**Comment by plbossart** [comment]
> > @libinyang can you look at the errors and add the configurations to alsa-tests so that the ucm-validator works?
> 
> Yes. I will do it. Just a silly question: How can I get the information in json file of the card, such as:
> 
> ```
> {
>     "sof-hda-dsp-1": {
>         "comment":	"Dummy test configuration for sof-hda-dsp",
>         "id":		"sof-hda-dsp",
>         "driver":	"sof-hda-dsp",
>         "name":		"sof-hda-dsp",
>         "longname":	"sof-hda-dsp",
>         "components":   "cfg-dmics:2",
>         "control": [
>             { "id": "iface=CARD,name='Headphone Mic Jack'" },
>             { "id": "name='Input Source'" }
>         ]
>     }
> }
> ```

It's not complicated, just look for the alsa-test results

````
ERR: sof-hda-dsp/sof-hda-dsp.conf: 'If'.'Dmic010'.'Condition' - True block not executed
````
and identify which control is used for this condition, and list it in the dummy.json file. If it's a regular mixer you don't need the iface=MIXER, it's implicit. the iface=CARD is only for jacks.

**Comment by libinyang** [comment]
> > > @libinyang can you look at the errors and add the configurations to alsa-tests so that the ucm-validator works?
> > 
> > 
> > Yes. I will do it. Just a silly question: How can I get the information in json file of the card, such as:
> > ```
> > {
> >     "sof-hda-dsp-1": {
> >         "comment":	"Dummy test configuration for sof-hda-dsp",
> >         "id":		"sof-hda-dsp",
> >         "driver":	"sof-hda-dsp",
> >         "name":		"sof-hda-dsp",
> >         "longname":	"sof-hda-dsp",
> >         "components":   "cfg-dmics:2",
> >         "control": [
> >             { "id": "iface=CARD,name='Headphone Mic Jack'" },
> >             { "id": "name='Input Source'" }
> >         ]
> >     }
> > }
> > ```
> 
> It's not complicated, just look for the alsa-test results
> 
> ```
> ERR: sof-hda-dsp/sof-hda-dsp.conf: 'If'.'Dmic010'.'Condition' - True block not executed
> ```
> 
> and identify which control is used for this condition, and list it in the dummy.json file. If it's a regular mixer you don't need the iface=MIXER, it's implicit. the iface=CARD is only for jacks.

@plbossart Thanks. So the dummy.json is manually filled, and the content is not auto generated. Is my understanding right?

**Comment by libinyang** [comment]
Confirmed that we will never use 'Dmic0 10' anymore based on the patch https://github.com/thesofproject/sof/pull/2942 and no customer platforms are using 'Dmic0 10'. So we can safely remove it.

File: ucm2/sof-hda-dsp/sof-hda-dsp.conf [hunk 1]
```diff
@@ -4,3 +4,26 @@ SectionUseCase."HiFi" {
 	File "HiFi.conf"
 	Comment "Play HiFi quality Music"
 }
+
+# the kcontrols initial values, which will be set by `alsactl init`
+
+BootSequence [
+	cset "name='Master Playback Switch' on"
+	cset "name='Master Playback Volume' 80"
+	cset "name='Capture Switch' on"
```
Comment (plbossart) [inline ucm2/sof-hda-dsp/sof-hda-dsp.conf:13]: @libinyang do you really want the switches on by default, or should they be set when an use cases needs capture or playback?

**Comment by libinyang** [comment]
Confirmed that pulseaudio can control these switches. Patch is updated based on Pierre's review.

**Comment by perexg** [comment]
Merged to 450bebd628975ffab153978b80f1493100910e53 .

**Pull request closed without merging**